garage/tf/algos/reps.py

```diff
@@ -0,0 +1,442 @@
+"""Relative Entropy Policy Search."""
+
+import numpy as np
+import scipy.optimize
+import tensorflow as tf
+
+from garage.misc import logger
+from garage.misc.overrides import overrides
+from garage.tf.algos import BatchPolopt
+from garage.tf.misc import tensor_utils
+from garage.tf.misc.tensor_utils import filter_valids
+from garage.tf.misc.tensor_utils import filter_valids_dict
+from garage.tf.misc.tensor_utils import flatten_batch
+from garage.tf.misc.tensor_utils import flatten_batch_dict
+from garage.tf.misc.tensor_utils import flatten_inputs
+from garage.tf.misc.tensor_utils import graph_inputs
+from garage.tf.optimizers import LbfgsOptimizer
+
+
+class REPS(BatchPolopt):
+    """
+    Relative Entropy Policy Search.
+
+    References
+    ----------
+    [1] J. Peters, K. Mulling, and Y. Altun, "Relative Entropy Policy Search,"
+        Artif. Intell., pp. 1607-1612, 2008.
+
+    Example:
+        $ python garage/examples/tf/reps_gym_cartpole.py
+
+    Args:
+        epsilon: dual func parameter.
+        l2_reg_dual: coefficient for dual func l2 regularization.
+        l2_reg_loss: coefficient for policy loss l2 regularization.
+        optimizer: policy optimizer.
+        optimizer_args: arguments of the policy optimizer.
+        dual_optimzier: dual func optimizer.
+        dual_optimizer_args: arguments of the dual optimizer.
+
+    """
+
+    def __init__(self,
+                 epsilon=0.5,
+                 l2_reg_dual=0.,
+                 l2_reg_loss=0.,
+                 optimizer=LbfgsOptimizer,
+                 optimizer_args=dict(max_opt_itr=50),
+                 dual_optimizer=scipy.optimize.fmin_l_bfgs_b,
```

Comment:
It is feasible to use LbfgsOptimizer instead? Otherwise, we have a divergent interface for optimizer against the other one above.

Reply:
No. I did find it is possible